api/src/main/java/org/apache/iceberg/view/UpdateViewRepresentation.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import org.apache.iceberg.PendingUpdate;
+
+/**
+ * API for adding/removing/updating a view's representations. Updating a {@link ViewRepresentation}
+ * can be done by calling {@link #remove(String)} and then {@link #add(ViewRepresentation)} for a
+ * particular dialect.
+ *
+ * <p>Apply returns the updated {@link ViewVersion} for validation.
+ *
+ * <p>When committing, these changes will be applied to the current view metadata. Commit conflicts
+ * will be resolved by applying the pending changes to the new view metadata.
+ */
+public interface UpdateViewRepresentation extends PendingUpdate<ViewVersion> {
+
+  /**
+   * Add a {@link ViewRepresentation} to the view.
+   *
+   * @param viewRepresentation The {@link ViewRepresentation} to add.
+   * @return this for method chaining
+   */
+  UpdateViewRepresentation add(ViewRepresentation viewRepresentation);
+
+  /**
+   * Remove a view representation for the given dialect.
+   *
+   * @param dialect The dialect for which the {@link ViewRepresentation} should be removed.
+   * @return this for method chaining
+   */
+  UpdateViewRepresentation remove(String dialect);
```

Comment:
With this semantics, we are basically treating `dialect` as the unique identifier of a representation within a list of representations. Should we make that explicit in the spec?

Reply:
And should we also prevent adding views with duplicated dialect name? We today don't check that in https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/view/ViewVersionParser.java#L78

Reply:
I've handled this in https://github.com/apache/iceberg/pull/8639